File: app/src/main/java/com/fyp/agrifarm/app/youtube/db/VideoDao.java
package com.fyp.agrifarm.app.youtube.db;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import java.util.List;

@Dao
public interface VideoDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insert(ShortVideo video);

    @Query("DELETE FROM " + ShortVideo.TABLE_NAME)
    void deleteAll();

    @Query("SELECT * FROM " + ShortVideo.TABLE_NAME)
    LiveData<List<ShortVideo>> getAllVideos();
}

File: app/src/main/java/com/fyp/agrifarm/app/youtube/db/VideoDao.kt
package com.fyp.agrifarm.app.youtube.db

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query

@Dao
interface VideoDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insert(vararg video: ExtendedVideo)

    @Query("DELETE FROM " + ExtendedVideo.TABLE_NAME)
    fun deleteAll()

    @get:Query("SELECT * FROM " + ExtendedVideo.TABLE_NAME)
    val allVideos: List<ExtendedVideo>
}